pkg/gpu/stream.go

```diff
@@ -8,21 +8,31 @@
 package gpu
 
 import (
+	"fmt"
 	"math"
 
+	model "github.com/DataDog/datadog-agent/pkg/collector/corechecks/gpu/model"
 	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/gpu/cuda"
 	gpuebpf "github.com/DataDog/datadog-agent/pkg/gpu/ebpf"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
 // StreamHandler is responsible for receiving events from a single CUDA stream and generating
 // kernel spans and memory allocations from them.
 type StreamHandler struct {
-	kernelLaunches []gpuebpf.CudaKernelLaunch
+	key            streamKey
+	kernelLaunches []enrichedKernelLaunch
 	memAllocEvents map[uint64]gpuebpf.CudaMemEvent
 	kernelSpans    []*kernelSpan
 	allocations    []*memoryAllocation
 	processEnded   bool // A marker to indicate that the process has ended, and this handler should be flushed
+	sysCtx         *systemContext
+}
+
+type enrichedKernelLaunch struct {
```

Comment:
what is the purpose of enrichedKernelLaunch struct? document it

Reply:
To avoid modifying the `gpuebpf.CudaKernelLaunch` struct which is shared with C code. Added the comment.